src/test/integration/savedItemsService.integration.ts

```diff
@@ -79,4 +79,36 @@ describe('SavedItemsService', () => {
     expect(savedItems[0].url).to.equal('https://abc');
     expect(savedItems[1].url).to.equal('https://def');
   });
+
```

Comment:
Just in case, would be nice to have the edge case where your offset is past the end of the list, since your while condition depends on that